crates/usvg/src/text/layout.rs

```diff
@@ -1418,7 +1418,7 @@ fn shape_text_with_font(
 
                 glyphs.push(Glyph {
                     byte_idx: ByteIndex::new(idx),
-                    cluster_len: end.checked_sub(start).unwrap_or(0), // TODO: can fail?
+                    cluster_len: end.saturating_sub(start), // TODO: can fail?
```

Comment:
The semantics are the same, but the meaning this suggests to the reader is different

Reply:
I don't know the history of this line, but seems like a runtime panic was too risky. Maybe a `debug_assert` above the push asserting `start <= end` would be nice here?

Reply:
Exactly, that's my reading as well.